src/Build/Graph/GraphBuilder.cs

```diff
@@ -495,24 +496,68 @@ void AddGraphBuildGlobalVariable(PropertyDictionary<ProjectPropertyInstance> glo
             var globalProperties = configurationMetadata.GlobalProperties.ToDictionary();
-
-            var projectInstance = _projectInstanceFactory(
+            ProjectGraphNode graphNode;
+            ProjectInstance projectInstance;
+           
+            if (!enableDynamicPlatformResolution)
+            {
+                projectInstance = _projectInstanceFactory(
                 configurationMetadata.ProjectFullPath,
                 globalProperties,
                 _projectCollection);
+                var dynamic = ConversionUtilities.ValidBooleanTrue(projectInstance.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName));
+                // We need this second evauation if the project is the first in a chain with setplatform negotitation as it needs to passdown its platform and its lookup table
+                if (ConversionUtilities.ValidBooleanTrue(projectInstance.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName)))
+                {
+                    globalProperties["platform"] = projectInstance.GetPropertyValue(PlatformMetadataName);
+                    globalProperties[PlatformLookupTableMetadataName] = projectInstance.GetPropertyValue(PlatformLookupTableMetadataName);
+                    projectInstance = _projectInstanceFactory(
+                                        configurationMetadata.ProjectFullPath,
+                                        globalProperties,
+                                        _projectCollection);
+                }
+            }
+            else
+            {
+                projectInstance = _projectInstanceFactory(
+                configurationMetadata.ProjectFullPath,
+                null,
+                _projectCollection);
+            }
 
             if (projectInstance == null)
             {
                 throw new InvalidOperationException(ResourceUtilities.GetResourceString("NullReferenceFromProjectInstanceFactory"));
             }
 
-            var graphNode = new ProjectGraphNode(projectInstance);
+            if (enableDynamicPlatformResolution)
+            {
+                String platformLookupTable;
+                globalProperties.TryGetValue(PlatformLookupTableMetadataName, out platformLookupTable);
+                String CurrentProjectPlatform;
+                globalProperties.TryGetValue(PlatformMetadataName, out CurrentProjectPlatform);
+                var SelectedPlatform = PlatformNegotiation.GetNearestPlatform(projectInstance.GetPropertyValue(PlatformMetadataName), projectInstance.GetPropertyValue(PlatformsMetadataName), projectInstance.GetPropertyValue(PlatformLookupTableMetadataName), platformLookupTable, projectInstance.FullPath, CurrentProjectPlatform, true);
+
+                if (SelectedPlatform != null)
+                {
+                    globalProperties["platform"] = SelectedPlatform;
```

Comment:
Use `PlatformMetadataName ` instead of `"platform"`

Reply:
changed